esp/esp_driver/network_adapter/main/spi_slave_api.c

```diff
@@ -37,18 +37,21 @@ static const char TAG[] = "SPI_DRIVER";
 				((VAL)& SPI_DMA_ALIGNMENT_MASK))
 
 #if (CONFIG_ESP_SPI_CONTROLLER == 3)
+    /* VSPI in ref manual, SPI3_HOST in spi_types.h */
     #define ESP_SPI_CONTROLLER 2
     #define GPIO_MOSI 23
     #define GPIO_MISO 19
     #define GPIO_SCLK 18
     #define GPIO_CS    5
 #elif (CONFIG_ESP_SPI_CONTROLLER == 2)
+    /* HSPI in ref manual, SPI2_HOST in spi_types.h */
     #define ESP_SPI_CONTROLLER 1
     #define GPIO_MOSI 12
```

Comment:
Could you please fix MOSI and MISO pins as well?

Reply:
MISO and MOSI are correct, they don't need fixing.

Reply:
Hi Gary,
Thanks for your input.

As you correctly pointed, These are IO_MUX pin numbers where you get the lower latency for SPI operations.
Also, In case different than these pins used / other order used, will be used as GPIO_MUX, which also should work, but higher latency than IO_MUX pins will be observed.

If we refer this [SPI terminolgy](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/spi_slave.html?highlight=spi%20slave#terminology) MOSI and MISO.

MOSI refers to `D` whereas MISO refers to `Q`.

From [ESP32-WROVER datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-wrover_datasheet_en.pdf) -> 2.2 Pin Description (page number 4), I could note below pins and their function.

| SPI2 | Function | Pin Number |
|:------------:|:--------------:|:----------:|
| HSPID | MOSI | IO13 |
| HSPIQ | MISO | IO12 |
| HSPICLK | Clock | IO14 |
| HSPICS0 | Chip Select | IO15 |

Reply:
Ok, this D/Q terminology is just confusing, don't know why would anyone use that instead of MISO/MOSI.
Note that in the SPI terminology link you provided, it also says:
MOSI Master In, Slave Out, a.k.a. D
=> which is a bad copy/paste I guess
I'll update the MISO/MOSI